### PR TITLE
[chore][extension/httpforwarder] update to use generated status

### PR DESCRIPTION
**Pull request by atoulme**
This updates the metadata for the extension to include stability information used to generate the status table.

### CODE REVIEW

File: extension/httpforwarder/README.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
 # HTTP Forwarder Extension
-
+<!-- status autogenerated section -->
 | Status                   |           |
 | ------------------------ |-----------|
-| Stability                | [beta]    |
+| Stability                | [beta]   |
+| Supported pipeline types |    |
```
Comment (mx-psi) [inline extension/httpforwarder/README.md:6]: I think we should make `mdatagen` skip this if `type` is `extension`. Wdyt?

**Review by mx-psi — APPROVED** [review]
LGTM, left a comment that can be addressed separately

File: extension/httpforwarder/README.md (hunk 1 above)
Comment (atoulme) [reply extension/httpforwarder/README.md:6]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20236 is open.

Comment (mx-psi) [reply extension/httpforwarder/README.md:6]: Merged that one :) can you rebase and regenerate this?

Comment (mx-psi) [reply extension/httpforwarder/README.md:6]: Merged that one :) can you rebase and regenerate this?

**Review by codeboten — APPROVED** [review]

File: extension/httpforwarder/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: httpforwarder
+
+status:
+  type: extension
+  stability: beta
+  pipelines: []
```
Comment (codeboten) [inline extension/httpforwarder/metadata.yaml:6]:
```suggestion
```

**Pull request merged**